core/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java

```diff
@@ -395,4 +408,10 @@ private static void checkLucene() {
         }
     }
 
+    static void cleanTmpFiles(Path tmpFile) throws IOException {
```

Comment:
I am not sure we should try to clean up stuff in here. We didn't do this for the `java.nio.tmpdir` so I am not sure we should do this here.